src/main/java/jenkins/plugins/slack/webhook/CommandRouter.java

```diff
@@ -0,0 +1,125 @@
+package jenkins.plugins.slack.webhook;
+
+
+import java.util.List;
+import java.util.ArrayList;
+
+import java.util.regex.Pattern;
+import java.util.regex.Matcher;
+
+import java.lang.reflect.Method;
+
+import jenkins.plugins.slack.webhook.exception.CommandRouterException;
+import jenkins.plugins.slack.webhook.exception.RouteNotFoundException;
+
+import jenkins.plugins.slack.webhook.model.SlackTextMessage;
+
+
+
+
+public class CommandRouter<T> {
+
+    public CommandRouter() { }
+
+    public List<Route> routes = new ArrayList<Route>();
+
+    public CommandRouter addRoute(String regex,
+        String command,
+        String commandDescription,
+        RouterCommand routerCommand) {
```

Comment:
Looking better, but needs some additional "parameterization" for the generics. Reasonably easy with an IDE to assist you. Here's a diff to look at: https://gist.github.com/OwensCode/b92fd2e29d0d275a4127